### PR TITLE
[ast] Reword some pgd/pok blocks in AST

### LINKED ISSUE #19556: [hw] AST sources fail Verible's "always-comb" lint rule
### Description

In #19426, I found that Verible warns that SV sources in `hw/top_earlgrey/ip/ast/` fail the [always-comb](https://chipsalliance.github.io/verible/verilog_lint.html#always-comb) lint rule.
  
Steps to reproduce:

1. If it hasn't been merged, check out #19426.
2. Remove the "always-comb" suppression from `hw/lint/tools/veriblelint/lowrisc-styleguide.rules.verible_lint`.
3. Run `util/dvsim/dvsim.py --tool=veriblelint hw/top_earlgrey/lint/top_earlgrey_lint_cfgs.hjson`

Relevant warnings:
```
../src/lowrisc_systems_ast_0.1/rtl/vcc_pgd.sv:34:1-6: Use 'always_comb' instead of 'always @*'. [Style: combinational-logic] [always-comb]

../src/lowrisc_systems_ast_0.1/rtl/vio_pgd.sv:34:1-6: Use 'always_comb' instead of 'always @*'. [Style: combinational-logic] [always-comb]

../src/lowrisc_systems_ast_0.1/rtl/vcaon_pgd.sv:34:1-6: Use 'always_comb' instead of 'always @*'. [Style: combinational-logic] [always-comb]

../src/lowrisc_systems_ast_0.1/rtl/vcmain_pgd.sv:34:1-6: Use 'always_comb' instead of 'always @*'. [Style: combinational-logic] [always-comb]
```

### PR DESCRIPTION
**Pull request by rswarbrick**
This code is in an ifndef synthesis block and works by asserting a
signal (e.g. vcc_pok_o) shortly after the start of time, using a
"#RDLY" delay to get the timing right.

Using "always @*" to structure this causes Verible to warn because of
its "always-comb" lint rule. There's no real reason to use this style
here (and the code goes back to the initial AST import in 9ee56b2 from
2020).

Make the ordering more explicit, and stick everything in the same
initial block: it should be easier to understand and does the same
thing.

Fixes #19556.

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]
LGTM from my end, but let's wait for approval from @meinsere and/or @arnonsha as well.

**Comment by msfschaffner** [comment]
CHANGE AUTHORIZED: hw/top_earlgrey/ip/ast/rtl/vcaon_pgd.sv
CHANGE AUTHORIZED: hw/top_earlgrey/ip/ast/rtl/vcc_pgd.sv
CHANGE AUTHORIZED: hw/top_earlgrey/ip/ast/rtl/vcmain_pgd.sv
CHANGE AUTHORIZED: hw/top_earlgrey/ip/ast/rtl/vio_pgd.sv

**Comment by meisnere** [comment]
The suggested change remove the ability to set different rise/fall delays.
Shouldn't replacing the "always (* xprop_off *) @( * )" of the existing logic with "always_comb (* xprop_off *)" solves the issue?

**Comment by arnonsha** [comment]
Also remember these are analog behavioral models. It's quite hard or even impossible to comply with digital design coding rules in such blocks.

**Comment by rswarbrick** [comment]
Right. I *think* that the latest version should fix those problems. Sorry for not noticing them!

I'm still a little confused, though. Take `vcaon_pgd.sv` for example. I'm trying to work out how the FDLY wait actually does anything. I think we need to be in a situation where `gen_supp_a` has been set to zero. But then (independent of my change), I think the output gets set to zero at the start of time and then set to zero again after VCAON_POK_FDLY. This won't do much! Is the point that we configure this by setting `gen_supp_a` to zero and also editing the initial block to set the output to `1'b1`?

**Comment by meisnere** [comment]
The logic is changed again, from continuous assignment to non-continuous (the initial block will check gen_supp_a only once).\
Why do you want to change it? Won't always_comb solve this issue?

**Comment by rswarbrick** [comment]
Oh, I see! So the point is that this block should be sensitive to backdoor changes to the local variable sometime in the middle of the simulation? This wasn't clear to me before!

And you're right that my proposed change has the properties you describe. Is each of these blocks supposed to act like a delay line between backdoor writes to `gen_supp_a` and a change to their output variable (`*_pok_o`)?

I'd like to get the "spec" straight. Even if we don't touch the code, the properties that it is supposed to have should probably be written down somewhere :-)

**Comment by rswarbrick** [comment]
This change looks like it has died, and I'm not convinced it's worth trying to resurrect :-) I'm sure we'll think about any associated lint problem again some time in the future!

**Pull request closed without merging**
